docs/retries.md

```diff
@@ -0,0 +1,408 @@
+                httpx.ConnectError,
+                httpx.ReadError
+            )),
+            wait=wait_exponential(multiplier=1, max=10),
+            stop=stop_after_attempt(3),
+            reraise=True
+        )
+    )
+    return httpx.AsyncClient(transport=transport)
+
+# Example usage
+client = create_network_resilient_client()
+# Client will now retry on timeout, connection, and read errors
+```
+
+### Custom Retry Logic
+
+```python {title="custom_retry_logic.py"}
+import httpx
+from tenacity import AsyncRetrying, wait_exponential, stop_after_attempt
+from pydantic_ai.tenacity import AsyncTenacityTransport, wait_retry_after
+
+def create_custom_retry_client():
+    """Create a client with custom retry logic."""
+    def custom_retry_condition(exception):
+        """Custom logic to determine if we should retry."""
+        if isinstance(exception, httpx.HTTPStatusError):
+            # Retry on server errors but not client errors
+            return 500 <= exception.response.status_code < 600
+        return isinstance(exception, (httpx.TimeoutException, httpx.ConnectError))
+
+    transport = AsyncTenacityTransport(
+        controller=AsyncRetrying(
+            retry=custom_retry_condition,
+            # Use wait_retry_after for smart waiting on rate limits,
+            # with custom exponential backoff as fallback
+            wait=wait_retry_after(
+                fallback_strategy=wait_exponential(multiplier=2, max=30),
+                max_wait=120
+            ),
+            stop=stop_after_attempt(5),
+            reraise=True
+        ),
+        validate_response=lambda r: r.raise_for_status()
+    )
+    return httpx.AsyncClient(transport=transport)
+
+client = create_custom_retry_client()
+# Client will retry server errors (5xx) and network errors, but not client errors (4xx)
+```
+
+## Using with Different Providers
+
+The retry transports work with any provider that accepts a custom HTTP client:
+
+### OpenAI
+
+```python {title="openai_with_retries.py"}
+from httpx import AsyncClient, HTTPStatusError
+from tenacity import AsyncRetrying, stop_after_attempt, retry_if_exception_type
+from pydantic_ai import Agent
+from pydantic_ai.models.openai import OpenAIModel
+from pydantic_ai.providers.openai import OpenAIProvider
+from pydantic_ai.tenacity import AsyncTenacityTransport, wait_retry_after
+
+def create_retrying_client():
+    """Create a client with retry functionality."""
+    def should_retry_status(response):
+        if response.status_code in (429, 502, 503, 504):
+            response.raise_for_status()
+
+    transport = AsyncTenacityTransport(
+        controller=AsyncRetrying(
+            retry=retry_if_exception_type(HTTPStatusError),
+            wait=wait_retry_after(max_wait=300),
+            stop=stop_after_attempt(5),
+            reraise=True
+        ),
+        validate_response=should_retry_status
+    )
+    return AsyncClient(transport=transport)
+
+client = create_retrying_client()
```

Comment:
We can reduce the duplication between these examples by adding `requires="smart_retry_example.py"` to the code fence, and then `from smart_retry_example import create_retrying_client`